### PR TITLE
Added warnings about XPath injection

### PR DESCRIPTION
**Pull request by zaichenko**
Hello!
I added search for XPath injection in several methods and changed XPathInjectionAnalyzerTest.

### CODE REVIEW

File: SecurityCodeScan.Test/Tests/Taint/XPathInjectionAnalyzerTest.cs [hunk 1]
```diff
@@ -17,90 +17,89 @@ protected override IEnumerable<DiagnosticAnalyzer> GetDiagnosticAnalyzers(string
...
-        Dim doc As New XmlDocument()
-        doc.Load("" / secret_config.xml"")
-        doc.SelectNodes(""/Config/Devices/Device[id='"" & input & ""']"")
-        doc.SelectSingleNode(""/Config/Devices/Device[type='"" & input & ""']"")
-    End Sub
-End Class
+            sink = sink.Replace("null", "Nothing").Replace("new ", "New ");
+
+            var visualBasicTest = $@"
+#Disable Warning BC50001
```
Comment (JarLob) [inline SecurityCodeScan.Test/Tests/Taint/XPathInjectionAnalyzerTest.cs]: Please remove trailing whitespace. (In Visual Studio: Edit->Advance->View White Space)

File: SecurityCodeScan/Config/Main.yml [hunk 2]
```diff
@@ -789,6 +789,68 @@ Sinks:
...
+
+  XPathNavigator_Evaluate_2:
+    Namespace: System.Xml.XPath
+    ClassName: XPathNavigator
+    Member: method
+    ArgTypes: (System.String, System.Xml.IXmlNamespaceResolver)
+    Name: Evaluate
+    InjectableArguments: [0]
+    Locale: SCS0003
+
```
[the remark this answers is not in the document]
Comment (JarLob) [reply SecurityCodeScan/Config/Main.yml]: I see also three more extension methods here https://docs.microsoft.com/en-us/dotnet/api/system.xml.xpath.extensions.xpathselectelement?view=netframework-4.7.2 to add.

**Comment by zaichenko** [comment]
I did it.

**Comment by JarLob** [comment]
Thanks! I have made some fixes to better support extension methods.

**Comment by JarLob** [comment]
Closes #24

**Pull request merged**